src/gui/src/abstractpreview.cpp

```diff
@@ -0,0 +1,1575 @@
+  QSize pageSizeScaled; // с учетом масштаба
+
+  QVector<QRect> marginRects;
+  int w, h; //высота и ширина рабочей области с учётом округления
+  //Отступы страницы и расстояние между страницами
+  int pageMargin, interPageSpacing;
+  int pageLabelHeight;
+  QRect pageLabelRect;
+
+  // отступы на листе в мм
+  qreal mm_left, mm_right, mm_top, mm_bottom;
+  // то же, но в пикселах
+  int mp_left, mp_right, mp_top, mp_bottom;
+  // с учетом масштаба
+  int ms_left, ms_top;
+  //  int	  ms_right,
+  //    	  ms_bottom;
+  //Общее количество страниц, колонок и строк страниц
+  int cols, rows;
+  //Параметры для передвижения страниц мышкой
+  QMousePressData mousePressData;
+  QPoint scrollBarValueOnMousePress;
+  //Мастштаб
+  double scale, scaleStep, scaleMax, scaleMin;
+
+  //Координаты для начальной страницы
+  qreal xx, yy;
+  int offsXpage;
+  int offsYpage;
+
+  //Необходимое количество строк страниц для отрисовки и номер начальной
+  //страницы с которой начинается отрисовка документа
+  int countPagesRow, beginPageNumber;
+
+  QList<QPageData *> pagesData;
+
+  AbstractPreview::PreviewMode viewMode;
+  int selectPageIndex;
+  QPoint topLeftPage;
+  QVector<QPageInfo> pagesRect;
+  QRect currentPageRect;
+  QPageInfo enterPageInfo;
+
+  bool mouseTrackingPage;
+  bool mouseEnterLeavePage;
+
+  bool movedContents;
+};
+//--------------------------------------------------------------
+
+TilesCache::TilesCache() {
+  maxWidth = TILE_MAX_WIDTH;
+  maxHeight = TILE_MAX_HEIGHT;
+  maxTiles = CACHE_SIZE_MAX / maxWidth / maxHeight;
+#ifndef QT_NO_DEBUG
+  qDebug() << "maxTiles: " << maxTiles;
+#endif
+}
+
+TilesCache::~TilesCache() { clear(); }
+
+void TilesCache::clear() {
+#ifndef QT_NO_DEBUG
+  qDebug() << "TilesCache::clear()";
+#endif
+  for (int i = tiles.size(); i > 0; i--)
+    delete tiles.takeLast();
+  tiles.clear();
+}
+
+void TilesCache::setTileSize(int maxW, int maxH) {
+#ifndef QT_NO_DEBUG
+  qDebug() << "TilesCache::setTileSize() " << maxW << "x" << maxH;
+#endif
+  clear();
+  maxWidth = maxW;
+  maxHeight = maxH;
+  maxTiles = CACHE_SIZE_MAX / maxWidth / maxHeight;
+#ifndef QT_NO_DEBUG
+  qDebug() << "maxTiles: " << maxTiles;
+#endif
+}
+
+void TilesCache::insertPages(int start, int count) {
+#ifndef QT_NO_DEBUG
+  qDebug() << "TilesCache::insertPages() start: " << start
+           << " count: " << count;
+#endif
+  for (int i = tiles.size() - 1; i >= 0; i--)
+    if (tiles[i]->page >= start)
+      tiles[i]->page += count;
+}
+
+void TilesCache::deletePages(int start, int count) {
+  int end = start + count - 1;
+#ifndef QT_NO_DEBUG
+  qDebug() << "TilesCache::deletePages() start: " << start
+           << " count: " << count;
+#endif
+  for (int i = tiles.size() - 1; i >= 0; i--) {
+    if (tiles[i]->page > end)
+      tiles[i]->page -= count;
+    else if (tiles[i]->page >= start)
+      delete tiles.takeAt(i);
+  }
+}
+
+QPixmap *TilesCache::getTile(int page, int xidx, int yidx) {
+  Tile *tile;
```

Comment:
Codacy found an issue: [The scope of the variable 'tile' can be reduced.](https://app.codacy.com/manual/jeremiahwilson777/qpxtool/pullRequest?prid=6191247)